aptos-move/aptos-release-builder/src/simulate.rs

```diff
@@ -0,0 +1,601 @@
+        .ok_or_else(|| anyhow!("failed to fetch gas schedule v2"))?;
+    let gas_feature_version = gas_schedule.feature_version;
+    let mut gas_params = AptosGasParameters::from_on_chain_gas_schedule(
+        &gas_schedule.into_btree_map(),
+        gas_feature_version,
+    )
+    .map_err(|err| {
+        anyhow!(
+            "failed to construct gas params at gas version {}: {}",
+            gas_feature_version,
+            err
+        )
+    })?;
+
+    // Overwrite the normal execution limits and write the patched gas schedule back.
+    let txn_gas_params = &mut gas_params.vm.txn;
+    txn_gas_params.max_execution_gas = txn_gas_params.max_execution_gas_gov;
+    txn_gas_params.max_io_gas = txn_gas_params.max_io_gas_gov;
+    txn_gas_params.max_storage_fee = txn_gas_params.max_storage_fee_gov;
+    state_view.set_state_value(
+        StateKey::resource(&AccountAddress::ONE, &StructTag {
+            address: AccountAddress::ONE,
+            module: Identifier::new(GasScheduleV2::MODULE_IDENTIFIER).unwrap(),
+            name: Identifier::new(GasScheduleV2::TYPE_IDENTIFIER).unwrap(),
+            type_args: vec![],
+        })?,
+        StateValue::new_legacy(
+            bcs::to_bytes(&GasScheduleV2 {
+                feature_version: gas_feature_version,
+                entries: gas_params.to_on_chain_gas_schedule(gas_feature_version),
+            })?
+            .into(),
+        ),
+    );
+
+    Ok(gas_params)
+}
+
+/***************************************************************************************************
+ * Simulation Workflow
+ *
+ **************************************************************************************************/
+fn list_scripts(proposal_path: &Path) -> Result<Vec<PathBuf>> {
+    let mut script_paths = std::fs::read_dir(proposal_path)?
+        .filter_map(|entry| entry.ok())
+        .filter_map(|entry| {
+            let path = entry.path();
+            if path.extension().map(|s| s == "move").unwrap_or(false) {
+                Some(path)
+            } else {
+                None
+            }
+        })
+        .collect::<Vec<_>>();
+
+    script_paths.sort();
+
+    Ok(script_paths)
+}
+
+pub async fn simulate_multistep_proposal(remote_url: Url, proposal_path: &Path) -> Result<()> {
+    // List all governance scripts.
+    let scripts = list_scripts(proposal_path)?;
+    if scripts.is_empty() {
+        bail!("no scripts found")
+    }
+
+    println!("Found {} scripts", scripts.len());
+    for path in &scripts {
+        println!("    {}", path.display());
+    }
+
+    // Compile all scripts.
+    println!("Compiling scripts...");
+    let mut compiled_scripts = vec![];
+    for path in &scripts {
+        let framework_package_args = FrameworkPackageArgs::try_parse_from([
+            "dummy_executable_name",
+            "--framework-local-dir",
+            &aptos_framework_path().to_string_lossy(),
+            "--skip-fetch-latest-git-deps",
+        ])?;
```

Comment:
for the ? operator here, I would suggest using Anyhow::Context which can give a little more info on what we were doing / why we failed in case of failure

Currently a lot of our tooling does not use this and since it also does not panic we are left completely without a stack trace although this is pretty straightforward